lstchain/tools/lstchain_create_dl3_file.py

```diff
@@ -0,0 +1,201 @@
+from lstchain.reco.utils import filter_events, get_effective_time
+from lstchain.paths import run_info_from_filename, dl2_to_dl3_filename
+from lstchain.irf import create_event_list
+
+from pyirf.utils import calculate_source_fov_offset
+
+__all__ = ["DataReductionFITSWriter"]
+
+
+class DataReductionFITSWriter(Tool):
+    name = "DataReductionFITSWriter"
+    description = __doc__
+
+    input_dl2 = traits.Path(
+        help="Input data DL2 file", exists=True, directory_ok=False, file_ok=True
+    ).tag(config=True)
+
+    output_dl3_path = traits.Path(
+        help="DL3 output filedir", directory_ok=True, file_ok=False
+    ).tag(config=True)
+
+    add_irf = traits.Bool(
+        help="True for adding IRF fits file to the DL3 file",
+        default_value=True,
+    ).tag(config=True)
+
+    input_irf = traits.Path(
+        help="Compressed FITS file of IRFs",
+        exists=True,
+        directory_ok=False,
+        file_ok=True,
+    ).tag(config=True)
+
+    source_name = traits.Unicode(help="Name of Source").tag(config=True)
+
+    config_file = traits.Path(
+        help="Config file for selection cuts",
+        directory_ok=False,
+        file_ok=True,
+    ).tag(config=True)
+
+    overwrite = traits.Bool(
+        help="If True, overwrites existing output file without asking",
+        default_value=True,
+    ).tag(config=True)
+
+    provenance_log = traits.Path(
+        help="Path for the Provenance log",
+        directory_ok=False,
+    ).tag(config=True)
+
+    aliases = {
+        "input_dl2": "DataReductionFITSWriter.input_dl2",
+        "d": "DataReductionFITSWriter.input_dl2",
+        "output_dl3_path": "DataReductionFITSWriter.output_dl3_path",
+        "o": "DataReductionFITSWriter.output_dl3_path",
+        "add_irf": "DataReductionFITSWriter.add_irf",
+        "input_irf": "DataReductionFITSWriter.input_irf",
+        "irf": "DataReductionFITSWriter.input_irf",
+        "config_file": "DataReductionFITSWriter.config_file",
+        "conf": "DataReductionFITSWriter.config_file",
+        "source_name": "DataReductionFITSWriter.source_name",
+        "provenance_log": "DataReductionFITSWriter.provenance_log",
+        "prov": "DataReductionFITSWriter.provenance_log",
+    }
+
+    flags = {
+        "overwrite": (
+            {"DataReductionFITSWriter": {"overwrite": True}},
+            "overwrite output file",
+        )
+    }
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.data = None
+        self.filename_dl3 = None
+        self.run_number = None
+        self.effective_time = None
+        self.elapsed_time = None
+        self.cuts = None
+        self.events = None
+        self.gti = None
+        self.pointing = None
+        self.aeff2d = None
+        self.edisp2d = None
+        self.bkg2d = None
+        self.psf = None
+        self.hdulist = None
+        self.output_file = None
+
+    def setup(self):
```

Comment:
You should not have to reimplemt config file parsing. `Tool` does that for you already.

Reply:
Any reason why you needed to do this?

Reply:
This is specifically for the selection cuts config file, which by default I have stored as `lstchain/data/data_selection_cuts.json`. The user can provide a similar separate config file as well, but if it is not provided, it will look for this standard file.

Reply:
Please do not do this. Add a corresponding traitlet where needed so that this can be set in the config file for this tool.

You can add an example config (standard config) to `lstchain/data` but it should not be necessary to add any additional code to parse config files.

Reply:
Ok! Then, maybe I can just rename the existing traitlet `config_file` to `selection_cuts`. Will that be alright?

Reply:
No! That's not what I meant. I meant you should implement the traitlets so that there is no second config file.

Reply:
@maxnoe this is what you expected, right? I mention it in the description as to what I do now.